projects/inverse/index.js

```diff
@@ -0,0 +1,205 @@
+  "0x41D079ce7282d49bf4888C71B5D9E4A02c371F9B", // inDAI->YFI
+  "0x2dCdCA085af2E258654e47204e483127E0D8b277", // inDAI->ETH
+];
+
+// ask comptroller for all markets array
+async function getAllTokens(block) {
+  let tokens = (
+    await sdk.api.abi.call({
+      block,
+      target: comptroller,
+      params: [],
+      abi: abi["getAllMarkets"],
+    })
+  ).output;
+  return tokens.filter(function (token) {
+    return ignore.indexOf(token) === -1;
+  });
+}
+
+async function getAllUnderlying(block, tokens) {
+  let allUnderlying = (
+    await sdk.api.abi.multiCall({
+      block,
+      calls: _.map(
+        tokens.filter((token) => token !== anETH),
+        (token) => ({
+          target: token,
+        })
+      ),
+      abi: abi["underlying"],
+    })
+  ).output;
+
+  allUnderlying.push({
+    input: {
+      target: anETH,
+    },
+    success: true,
+    output: wETH,
+  });
+  return allUnderlying;
+}
+
+async function getCashes(block, tokens) {
+  return (
+    await sdk.api.abi.multiCall({
+      block,
+      calls: _.map(tokens, (token) => ({
+        target: token,
+      })),
+      abi: abi["getCash"],
+    })
+  ).output;
+}
+
+async function getTotalSupplies(block, tokens) {
+  return (
+    await sdk.api.abi.multiCall({
+      block,
+      calls: _.map(tokens, (token) => ({
+        target: token,
+      })),
+      abi: abi["totalSupply"],
+    })
+  ).output;
+}
+
+async function anchorTVL(block) {
+  const balances = {};
+
+  const tokens = await getAllTokens(block);
+  const [allUnderlying, cashes] = await Promise.all([
+    getAllUnderlying(block, tokens),
+    getCashes(block, tokens),
+  ]);
+
+  _.each(tokens, (token) => {
+    let cash = _.find(
+      cashes,
+      (result) => result.success && result.input.target === token
+    );
+    let underlying = _.find(
+      allUnderlying,
+      (result) => result.success && result.input.target === token
+    );
+    if (cash && underlying) {
+      balances[underlying.output] = BigNumber(
+        balances[underlying.output] || 0
+      ).plus(cash.output);
+    }
+  });
+
+  return balances;
+}
+
+async function vaultsTVL(block) {
+  const balances = {};
+
+  const [allUnderlying, totalSupplies] = await Promise.all([
+    getAllUnderlying(block, vaults),
+    getTotalSupplies(block, vaults),
+  ]);
+
+  _.each(vaults, (token) => {
+    let totalSupply = _.find(
+      totalSupplies,
+      (result) => result.success && result.input.target === token
+    );
+    let underlying = _.find(
+      allUnderlying,
+      (result) => result.success && result.input.target === token
+    );
+    if (totalSupply && underlying) {
+      balances[underlying.output] = BigNumber(
+        balances[underlying.output] || 0
+      ).plus(totalSupply.output);
+    }
+  });
+
+  return balances;
+}
+
+async function stabilizerTVL(block) {
+  const totalSupply = (
+    await sdk.api.abi.call({
+      block,
+      target: dola,
+      abi: abi["totalSupply"],
```

Comment:
Not all of the DOLA `totalSupply` originates from the Stabilizer. Some originate from Anchor. Instead, you should query `Stabilizer.supply()` which will return the amount of Dai stored within the Stabilizer.

Reply:
Also something to note is that DOLA in general should not be counted as part of the TVL since it's a synthetic asset that can be minted in any quantity within our system.

Reply:
Yeah that's why I wasn't sure what to use. I was using `Stabilizer.supply()` initially before I changed it to the `totalSupply`.